core/blockly.js

```diff
@@ -206,20 +206,23 @@ Blockly.svgResize = function(workspace) {
 Blockly.onKeyDown_ = function(e) {
   var mainWorkspace = Blockly.mainWorkspace;
 
-  if (mainWorkspace.options.readOnly &&
-      Blockly.keyboardAccessibilityMode &&
-      Blockly.navigation.onKeyPress(e)) {
-    return true;
-  }
-
-  if (mainWorkspace.options.readOnly || Blockly.utils.isTargetInput(e) ||
+  if (Blockly.utils.isTargetInput(e) ||
       (mainWorkspace.rendered && !mainWorkspace.isVisible())) {
-    // No key actions on readonly workspaces.
     // When focused on an HTML text input widget, don't trap any keys.
     // Ignore keypresses on rendered workspaces that have been explicitly
     // hidden.
     return;
   }
+
+  if (mainWorkspace.options.readOnly) {
+    // When in read only mode handle key actions for keyboard navigation.
+    if (Blockly.keyboardAccessibilityMode &&
+      Blockly.navigation.onKeyPress(e)) {
+      return true;
```

Comment:
This method has no return. You can just check keyboardAccessibilityMode and then call onKeyPress without the return true.